sklearn/preprocessing/label.py

```diff
@@ -53,6 +54,140 @@ def _check_numpy_unicode_bug(labels):
                            " NumPy to use LabelEncoder with unicode inputs.")
 
 
+class TargetTransformer(BaseEstimator):
+    """Meta-estimator to apply a transformation to the target before fitting
+
+    Useful for applying a non-linear transformation like np.log
+    to the target in a regression problem.
+
+    The computation during ``fit`` is::
+
+        estimator.fit(X, func(y))
+
+    The computation during ``predict`` is::
+
+        inverse_func(estimator.predict(X))
+
+
+    Parameter
+    ---------
+    estimator : object
+        Estimator object to wrap.
+
+    func : function
```

Comment:
Are there cases in which it is useful to have a transformer here, i.e. one of:
* The transformation involves statistics from the training portion of y
* The transformation is parameterised and those parameters should be searchable